src/dbnode/namespace/convert.go

```diff
@@ -212,6 +218,15 @@ func ToMetadata(
 	return NewMetadata(ident.StringID(id), mOpts)
 }
 
+// ToStagingState converts nsproto.StagingState to StagingState.
+func ToStagingState(state *nsproto.StagingState) (StagingState, error) {
+	if state == nil {
+		return nil, nil
+	}
+
+	return NewStagingState(Status(state.Status))
```

Comment:
This should probably just pass in the `state.Status` and let `NewStagingState` deal with the conversions

Reply:
Good point. Will fix up